code/game/objects/structures/curtains.dm

```diff
@@ -9,17 +9,122 @@
+	var/assembled = FALSE
+	var/overlay_color = "#ffffff"
+	var/overlay_alpha = 255
+
+
+/obj/structure/curtain/Initialize(mapload)
+	. = ..()
+	if(opacity)
+		icon_state = "closed"
+		overlay = image("closed_overlay")
+		overlay.color = overlay_color
+		overlay.alpha = overlay_alpha
+		add_overlay(overlay)
+		layer = SHOWER_CLOSED_LAYER
+	else
+		icon_state = "open"
+		overlay = image("open_overlay")
+		overlay.color = overlay_color
+		overlay.alpha = overlay_alpha
+		add_overlay(overlay)
+		layer = SHOWER_OPEN_LAYER
 
 /obj/structure/curtain/open
 	icon_state = "open"
 	layer = SHOWER_OPEN_LAYER
 	opacity = FALSE
 
+/obj/item/mounted/curtain/curtain_fixture
+	icon_state = "handheld"
+	icon = 'icons/obj/curtain.dmi'
+	name = "\improper curtain rod assembly"
+	new_attack_chain = TRUE
+
+/obj/item/mounted/curtain/curtain_fixture/interact_with_atom(atom/target, mob/living/user, list/modifiers)
+	. = ..()
+	if(!istype(target,/obj/structure/window) && !istype(target,/turf/simulated/wall/))
+		return
+	var/on_wall = get_turf(target)
+	to_chat(user, "<span class='notice'>You begin attaching [src] to the [on_wall].</span>")
+	playsound(get_turf(src), 'sound/machines/click.ogg', 75, TRUE)
+	if(!do_after(user, 3 SECONDS, target = on_wall))
+		return
+	var/obj/structure/curtain/assembly/new_curtain = new /obj/structure/curtain/assembly(on_wall)
+	new_curtain.fingerprints = src.fingerprints
+	new_curtain.fingerprintshidden = src.fingerprintshidden
+	new_curtain.fingerprintslast = src.fingerprintslast
+
+	user.visible_message("<span class='notice'>[user] attaches the [src] to the [on_wall].</span>", \
+		"<span class='notice'>You attach the [src] to the [on_wall].</span>")
+	qdel(src)
+
+/obj/item/mounted/curtain/curtain_fixture/activate_self(mob/user)
+	. = ..()
+	to_chat(user, "<span class='notice'>You begin attaching [src] to the ceiling.</span>")
+	playsound(get_turf(src), 'sound/machines/click.ogg', 75, TRUE)
+	if(!do_after(user, 3 SECONDS, target = get_turf(user)))
+		return
+	var/obj/structure/curtain/assembly/new_curtain = new /obj/structure/curtain/assembly(get_turf(user))
+	new_curtain.fingerprints = src.fingerprints
+	new_curtain.fingerprintshidden = src.fingerprintshidden
+	new_curtain.fingerprintslast = src.fingerprintslast
+
+	user.visible_message("<span class='notice'>[user] attaches the [src] to the ceiling.</span>", \
+		"<span class='notice'>You attach the [src] to the ceiling.</span>")
+	qdel(src)
+
+
+
+/obj/structure/curtain/assembly
+	icon_state = "assembly0"
+	name = "Curtain Rod"
+	opacity = FALSE
+	density = FALSE
+	desc = "A curtain assembly! It needs a <b>material</b>."
+
+/obj/structure/curtain/assembly/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>Alt-Click to take it down.</span>"
+
+/obj/structure/curtain/assembly/AltClick(mob/user)
+	if(user.incapacitated())
+		to_chat(user, "<span class='warning'>You can't do that right now!</span>")
+		return
+	if(!Adjacent(user))
+		return
+	// TODO: turn back into assembly item
```

Comment:
TODO here, planning on doing it in this PR?

Reply:
Can remove the comment. Forgot to remove it, you can see it get turned back into a fixture assembly underneath.